Fix/headings in tables

diegosteiner opened the pull request:
When the markup contains a table with a nested heading inside, the current behavior will pass on all options of the heading to the cell, which can lead to an error like this:

```rb
undefined method `style=' for an instance of Prawn::Table::Cell::Text
```

Proposal: only pass on known options to prawn-table

codez commented on lib/prawn/markup/processor/tables.rb:
```diff
@@ -86,6 +86,11 @@ def inside_container?
+        options[:font_style] ||= options.delete(:style) if options.key?(:style)
+        # only allow on supported options of prawn-table; See https://github.com/prawnpdf/prawn-table/blob/master/manual/table/cell_text.rb
+        options.slice!(*%i[font font_style inline_format kerning leading min_font_size
+                           overflow rotate rotate_around single_line size text_color valign])
```
The options hash must not be modified in this method. Use `options = options.slice(...)` to create a new object. Furthermore, `slice!` is not in the ruby stdlib anyways.

codez reviewed (changes requested):
Thanks for the pull request, this looks good. Please check the action results for remaining issues.

diegosteiner commented:
I've fixed the specs and linting and changed to code as requested. 

I'd be glad if you could look into the specs, I was unable to figure out why the top_positions of the nested headings are where are now.

diegosteiner commented on spec/prawn/markup/processor/tables_spec.rb:
```diff
@@ -77,11 +76,26 @@
+      processor.parse('<table><tr><td><h1>hello</h1><h2>world</h2></td></tr></table>')
+      expect(text.strings).to eq(['hello', 'world'])
+      expect(left_positions).to eq([first_col_left, first_col_left])
+      expect(top_positions).to eq([first_row_top - 17, first_row_top - 50].map(&:round))
```
Why do the headings start at offset 17 and (36 + 14)?

codez replied:
Heading top positions are indeed complex, even in the headings specs we use the "magic" values. I'm fine with these hard-coded values here.

codez commented:
@diegosteiner FYI, I just released a new version of prawn-markup with these changes
